### DISCUSSION TITLE
Vercel and Microsoft SQL Connector

### DISCUSSION THREAD

**Discussion by marcelommss** [prisma/prisma#5934] [Q&A]
Hi,

I'm trying to deploy an next.js app with prisma.js using Microsoft SQL Server database.
In my local dev environment it's working fine, but when deployed to vercel, I'm getting this error:

```
Timed out fetching a new connection from the pool. Please consider reducing the number of requests or increasing the `connection_limit` parameter (https://www.prisma.io/docs/concepts/components/prisma-client/connection-management#connection-pool). Current limit: 200.
```

I significantly increased the connection limit, but I'm still getting error.

Any toughts what it may be?

**Comment by ryands17** [comment]
@marcelommss 👋 

Is your database in the same region where you are deploying your Next.js app? If not, then this could happen due to latency issues with the database being too far away.

**Reply by pimeys** [reply]
You could try a few things. First, on the SQL Server side all parameters are camelCased, and the correct one would be `connectionLimit`. I'm planning to document these the next thing, so it wouldn't be so confusing. Also the error message should point to these docs instead of printing `connection_limit` to the error.

Your problem is that queries take a long time. You have a limit of 200 connections, but what is your roundtrip time from Vercel to the SQL Server? 200 connections is a lot, but what is the rate of requests you're giving to the Prisma instance? You could always change `poolTimeout` to `0`, meaning we never timeout when the pool is over capacity, leading to an increased memory use instead of timeouts.

**Reply by marcelommss** [reply]
@ryands17 both are in US

**Reply by marcelommss** [reply]
@pimeys I tried setting the poolTimeout to 0, but got this error: Error opening a TLS connection: The TLS settings didn't allow the connection to be established. Please review your connection string. (error: Connection reset by peer (os error 104))

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
